roles/basic_setup_rhel/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+---
+- name: "Subscription Manifest"
+  include_role:
+    name: theforeman.foreman.manifest
+
+- name: "Enable RHEL7 repository"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux Server
+        repository_sets:
+          - name: Red Hat Enterprise Linux 7 Server (RPMs)
+            basearch: x86_64
+            releasever: 7Server
+  when: enable_rhel7
+
+- name: "Enable RHEL8 repositories"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux for x86_64
+        repository_sets:
+          - name: Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)
+            releasever: 8
+          - name: Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
+            releasever: 8
+  when: enable_rhel8
+
+- name: "Sync RHEL7 repository"
+  theforeman.foreman.repository_sync:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    organization: "{{ foreman_organization }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs }}"
+    product: Red Hat Enterprise Linux Server
+  async: "{{ 3600 if not wait_for_syncs else omit }}"
+  poll: "{{ 0 if not wait_for_syncs else omit }}"
+  when: enable_rhel7 and sync_rhel7_now
+
+- name: "Sync RHEL8 repositories"
+  theforeman.foreman.repository_sync:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    organization: "{{ foreman_organization }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs }}"
+    product: Red Hat Enterprise Linux for x86_64
+  async: "{{ 3600 if not wait_for_syncs else omit }}"
+  poll: "{{ 0 if not wait_for_syncs else omit }}"
+  when: enable_rhel8 and sync_rhel8_now
+
+- name: "Create Sync Plan"
+  include_role:
+    name: theforeman.foreman.sync_plans
+  vars:
+    foreman_sync_plans:
+      - name: "{{ sync_plan_name | default('RHEL Sync Plan') }}"
+        interval: "{{ sync_plan_interval | default(daily) }}"
+        cron_expression: "{{ sync_plan_cron_expression | default(omit) }}"
+        sync_date: "{{ sync_plan_sync_date | default('2020-01-01 00:00:00 UTC') }}"
+        products:
+          - Red Hat Enterprise Linux Server
+          - Red Hat Enterprise Linux for x86_64
+
+- name: "Create Activation Key"
+  include_role:
+    name: theforeman.foreman.activation_keys
+  vars:
+    foreman_activation_keys:
+      - name: "{{ activation_key_name | default('base_rhel_key') }}"
```

Comment:
this doesn't add any subs, thus assuming SCA is enabled?

Reply:
Auto-attach is enabled on activation keys by default, so it's not a strictly necessary assumption that SCA is enabled I think. The potential pitfall would be a subscription like 'Satellite Infrastructure Subscription' being consumed by RHEL machines because it also covers the RHEL product ID. But I believe auto-attach is supposed to prefer the minimal available subscription that covers all `/etc/pki/product/*.pem` product certificates present on the registering content host.